### PR TITLE
[eclipse/xtext#1777] Translate more code to java.

### PR DESCRIPTION
**Pull request by ArneDeutsch**
org.eclipse.xtext, org.eclipse.xtext.ide and org.eclipse.xtext.util are
Xtend free. Remov xtend-gen folders from build.

Signed-off-by: Arne Deutsch <Arne.Deutsch@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	public static final String UNKNOWN_SCOPE = "unknown.xtext";
+
+	/**
+	 * TextMate scopes indicating an unhandled style ID to TextMate scope mapping.
+	 */
+	public static final List<String> UNKNOWN_SCOPES = Collections
+			.singletonList(SemanticHighlightingRegistry.UNKNOWN_SCOPE);
+
+	@Inject
+	@Extension
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:70]: Not necessary in Java

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+			LanguageClient client) {
+		Preconditions.checkState(this.client == null, "Already initialized.");
+		TextDocumentClientCapabilities textDocument = capabilities == null ? null : capabilities.getTextDocument();
+		SemanticHighlightingCapabilities semanticHighlightingCapabilities = textDocument == null ? null
+				: textDocument.getSemanticHighlightingCapabilities();
+		boolean semanticHighlighting = semanticHighlightingCapabilities == null ? false
+				: semanticHighlightingCapabilities.getSemanticHighlighting();
+		ImmutableBiMap.Builder<Integer, List<String>> builder = ImmutableBiMap.builder();
+		if (semanticHighlighting) {
+			Set<List<String>> set = IterableExtensions.toSet(IterableExtensions.filter(
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:90]: Please either go for Stream API, FluentIterable or traditional for loops. But these inversely nested static methods make it really hard to understand the code.

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+		if (isNullOrUnknown(scopes))
+			return -1;
+		Integer index = this.scopes.inverse().get(scopes);
+		return index == null ? -1 : index;
+	}
+
+	/**
+	 * Returns with a view of all scopes known by this manager.
+	 */
+	public List<List<String>> getAllScopes() {
```
Comment (szarnekow) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:128]: API design at its best. So sad.

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java (hunk 1 above)
Comment (ArneDeutsch) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:70]: Removed annotation.

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java (hunk 2 above)
Comment (ArneDeutsch) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:90]: Yes, it is hard to understand. Will try to refactor it.

Comment (ArneDeutsch) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java:90]: Refactored to traditional for loops. Please verify that my implementation maps well to the old implementation.

Comment (ArneDeutsch) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java]: Seems our tests are not THAT bad ... seems they catched a bug in my implementation ... will check.

Comment (ArneDeutsch) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/semanticHighlight/SemanticHighlightingRegistry.java]: Have rewritten it using Streams for a more direct mapping to the old code and to fix the tests. Please review again.

File: org.eclipse.xtext.util/src/org/eclipse/xtext/util/internal/EmfAdaptableProcessor.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+import org.eclipse.emf.common.notify.impl.AdapterImpl;
+import org.eclipse.xtend.lib.macro.AbstractClassProcessor;
+import org.eclipse.xtend.lib.macro.RegisterGlobalsContext;
+import org.eclipse.xtend.lib.macro.TransformationContext;
+import org.eclipse.xtend.lib.macro.declaration.ClassDeclaration;
+import org.eclipse.xtend.lib.macro.declaration.MutableClassDeclaration;
+import org.eclipse.xtend2.lib.StringConcatenationClient;
+import org.eclipse.xtext.xbase.lib.Extension;
+
+public class EmfAdaptableProcessor extends AbstractClassProcessor {
```
Comment (szarnekow) [inline org.eclipse.xtext.util/src/org/eclipse/xtext/util/internal/EmfAdaptableProcessor.java:24]: Rewritign AA to Java goes full circle I suppose :)

**Pull request merged**